packages/e2e/suites/full/helpers/dappwright.ts

```diff
@@ -152,7 +176,38 @@ export const test = baseTest.extend<{
     await use(metamask);
   },
 
-  switcher: async ({ wallet }, use) => {
-    await use(new RoleSwitcher(wallet));
+  switcher: async ({ wallet, page }, use) => {
+    const switcher = new RoleSwitcher(wallet);
+    const original = wallet.confirmTransaction;
+
+    // Each time there is a tx signature we save the address that produced the signature.
+    wallet.confirmTransaction = async () => {
+      switcher.pushToHistory();
+      await original.bind(wallet)();
+    }
+
+    await use(switcher);
+
+    // After each test we go through each address that has signed and
+    // Clear the tx nonce in metamask.
+    // This is because metamask doesn't automatically fix the nonce when it doesn't
+    // match with the network. Doing this forces metamask to refresh the nonce.
+    const history = switcher.popHistory()
+    for (const entry of history) {
+      await entry()
+
```

Comment:
How do we make sure we're not clearing the tx nonce more than one time on a given account? I think with the current `this.history` there's no easy way to remove duplicated entries, perhaps using a Set with an object like {role, number} could work.

Reply:
Clearing more than once procuce no issues except for "extra wait". I don't believe that's a problem. Specially because clearing it's really fast.